### ISSUE TITLE
SQS RedriveAllowPolicy

### ISSUE THREAD

**Issue by paulthorburn** [hashicorp/terraform-provider-aws#22249]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Hello,

I couldn't see anything in SQS resource documentation or in a State file (not sure where else to look) that would indicate the SQS RedriveAllowPolicy setting is configurable by terraform and was hoping to get it implemented.

### New or Affected Resource(s)

Resource: `aws_sqs_queue`

### Potential Terraform Configuration
 The RedriveAllowPolicy is jsonencode and very similar to redrive_policy which already exists in terraform today. I assume it would something like

```hcl
  RedriveAllowPolicy = jsonencode({
    redrivePermission = "allowAll"
  })

  RedriveAllowPolicy = jsonencode({
    redrivePermission = "byQueue" : [aws_sqs_queue.terraform_queue.arn]
  })

  RedriveAllowPolicy = jsonencode({
    redrivePermission = "denyAll" 
  })
```

### References
Cloudformation Attribute: RedriveAllowPolicy
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sqs-queues.html

**Comment by ewbankkit** [comment]
The `RedriveAllowPolicy` queue attribute was made available [AWS SDK v1.40.34](https://github.com/aws/aws-sdk-go/releases/tag/v1.40.34) - #20749.

**Comment by dirk39** [comment]
Let me add it
